src/Core/Grid/Presenter/GridPresenter.php

```diff
@@ -30,6 +30,7 @@
 use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\PositionColumn;
 use PrestaShop\PrestaShop\Core\Grid\Definition\GridDefinitionInterface;
 use PrestaShop\PrestaShop\Core\Grid\Filter\FilterInterface;
+use Symfony\Component\DependencyInjection\Container;
```

Comment:
Too bad we don't have another class to use to camelize?

Reply:
Nope, and this is what we are already using https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Grid/GridFactory.php#L92

Reply:
yes I know, it's just that we were completely framework agnostic until we use this class
not really a problem since it's integrated in the core anyway, but still my heart cries a little.. 🥺

Reply:
:broken_heart:

Reply:
> yes I know, it's just that we were completely framework agnostic until we use this class
not really a problem since it's integrated into the core anyway, but still my heart cries a little..

Me too, but it's not me but @sarjon who introduced it with the approval of @matks xD

In 1.7.7, we may improve that but for now, we need to provide consistent behavior, isn't it?

Reply:
> Too bad we don't have another class to use to camelize?

There is option to use `\Doctrine\Common\Inflector\Inflector::camelize()`, but it won't change much right?

> just that we were completely framework agnostic

We were not. :/ Grid already depends on Symfony Form component and Doctrine. In fact, `Core` is not framework agnostic at all. :disappointed: 

> Me too, but it's not me but @sarjon who introduced it with the approval of @matks

I don't think it's worth copy/paste 3rd party lib inside PS, what benefits would that bring? :thinking:

Reply:
> yes I know, it's just that we were completely framework agnostic until we use this class
> not really a problem since it's integrated in the core anyway, but still my heart cries a little.. 

We are framework agnostic. But we are not component agnostic.
In the Core code written for migration, you'll never find a class using `Symfony\Bundle`. But you'll find `Symfony\Component`. This means that if we want, we can ditch Symfony framework and replace it with Laravel framework. But we still rely on Symfony components (as does Laravel !).

I think this is a fair compromise between coupling matters and code simplicity matters. But if this becomes an issue, we can aim for a full 3rd-party independant Core and rely on interfaces being implemented by external libs, using hexagonal architecture.

I think it's too early to do it 😉 but I'm listening to feedbacks ;) .